designcompose/src/main/java/com/android/designcompose/DesignFrame.kt

```diff
@@ -623,9 +619,10 @@ internal fun DesignFrame(
         }
         is LayoutInfoAbsolute -> {
             if (parentLayout?.isWidgetChild == true) {
+                val childLayout = layout ?: LayoutManager.getGridNodeSize(name)
                 // For direct children of a widget, render the frame as a box with the calculated
                 // layout size, then compose the frame's children with our custom layout
-                Box(m.layoutSizeToModifier(layout)) {
+                Box(m.layoutSizeToModifier(childLayout)) {
```

Comment:
Pity we need a Box here - is there something different about how we report the size from `DesignFrameLayout` than how `Box` does it? I'm wondering about cases where the evaluated layout ends up being a different size than the design-tool computed size.

Reply:
This was really confusing when I implemented it and it's possible it's not needed anymore. I'll investigate.

Reply:
I've made some changes here such as removing the `Box()` under `DesignFrameLayout()` as it's not needed anymore, but without this particular `Box()`, some of the children are laid out incorrectly and I'm not sure why. For example, the child in the widget at the top is stretched to fit all four columns, and then the green square gets moved to the center instead of hugging the left like it should The bottom node is an example of how it renders outside of a grid layout.
![Screenshot 2023-11-03 at 4 53 00 PM](https://github.com/google/automotive-design-compose/assets/130500397/4dcee4ad-1259-4a64-9db1-0e9ccad62eb1)

Reply:
Hm, there's probably something else going on underneath that's not quite right, then... I'm guessing layout is generating the same output either way, so I'm confused what's going wrong (worst case, we have a bug where we don't show what layout generated correctly).